src/test/java/org/mozilla/jss/tests/CapabilitiesList.java

```diff
@@ -0,0 +1,260 @@
+package org.mozilla.jss.tests;
+
+import java.security.Security;
+import java.security.Provider;
+
+import java.util.*;
+import java.security.*;
+import javax.crypto.*;
+import java.security.Provider;
+import java.security.Security;
+import java.util.Enumeration;
+import java.util.Properties;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import org.mozilla.jss.CryptoManager;
+import org.mozilla.jss.InitializationValues;
+import org.mozilla.jss.JSSProvider;
+import org.mozilla.jss.util.Password;
+import org.mozilla.jss.util.PasswordCallback;
+import org.mozilla.jss.util.PasswordCallbackInfo;
+import org.mozilla.jss.util.NullPasswordCallback;
+import org.mozilla.jss.crypto.AlreadyInitializedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * List the available capabilities for ciphers, key agreement, macs, message
+ * digests, signatures and other objects for the Mozilla-JSS provider.
+ *
+ * The listing is done via two methods:
+ * 1) A brief enumeration from example given at page that can no longer be found
+ *    http://www.java2s.com/Code/Java/Security/ListAllProviderAndItsAlgorithms.html
+ * 2) A verbose enumeration based on example 1 from Cryptography for Java by David Hook
+ *
+ * Initialization code is like the one in org.mozilla.jss.tests.HmacTest
+ */
+public class CapabilitiesList {
+
+    // start Capabilities inner class
+
+    /**
+     * List the available capabilities for ciphers, key agreement, macs, message
+     * digests, signatures and other objects for the Mozilla-JSS provider.
+     *
+     * The listing is done via two methods:
+     * 1) A brief enumeration from example given at page that can no longer be found
+     *    http://www.java2s.com/Code/Java/Security/ListAllProviderAndItsAlgorithms.html
+     * 2) A verbose enumeration based on example 1 from Cryptography for Java by David Hook
+     *
+     * It incorporates code from org.mozilla.jss.tests.JSSProvider
+     */
+    static class Capabilities {
+
+        public static Logger logger = LoggerFactory.getLogger(CapabilitiesList.class);
+        public static String briefFileBase = "listings/brief/Capabilities4";
+        public static String verboseFileBase = "listings/verbose/Capabilities4";
+
+        /* Inner class to use existing system nss database
+         */
+        public static class UseSystemDB {
```

Comment:
If possible avoid using deeply nested classes. Could `UseSystemDB` be moved under `CapabilitiesList`?